testplan/web_ui/testing/src/AssertionPane/AssertionTypes/TextAttachment.js

```diff
@@ -0,0 +1,270 @@
+            </SyntaxHighlighter>
+        )
+
+        return return_jsx
+    }
+
+    /**
+    * Fetch the text attachment.
+    *   * Handle UID errors.
+    *   * Make a GET request for the .txt attachment file.
+    *   * Prepare the display content
+    * @public
+    */
+    getTextAttachment() {
+    const paths = window.location.pathname.split('/');
+
+    // Check if dev mode and display test_string
+    if ((paths.length >= 2) && (paths[1] === '_dev')) {
+        const TEST_TEXT = "test1\ntest2\ntest3\ntest4\ntest5\ntest6\ntest7\n";
+        let text = TEST_TEXT;
+        let lines = text.split('\n');
+        let length = lines.length;
+
+        if(lines.pop() === ""){
+            text = text + "<newline>"
+            lines[length-1] = "<newline>"
+        }
+
+        this.setState({
+                        originalText: text,
+                        numberOfLines: length
+                      });
+
+        let display = this.getCollapsedText(lines);
+
+        this.setState({
+          textContent: display,
+          loading: false
+        });
+
+    //If not in dev mode
+    } else if (paths.length >= 3) {
+        axios.get(this.props.src)
+        .then(response => {
+            let text = response.data
+            let lines = response.data.split('\n');
+            let length = lines.length;
+
+            if(lines.pop() === ""){
+                text = text + "<newline>"
+                lines[length-1] = "<newline>"
+             }
+
+            this.setState({
+                            originalText: text,
+                            numberOfLines: length
+                          });
+
+            let display = this.getCollapsedText(lines);
+
+            this.setState({
+              textContent: display,
+              loading: false
+            });
+        })
+        .catch(error => this.setState({
+                                          error: true,
+                                          loading: false
+                                      })
+              );
+
+      //If some other error with URL
+    } else {
+        this.setState({
+                        error: true,
+                        loading: false
+                      });
+      }
+    }
+
+    /*
+     * Change the displayed .txt content, when the Expand/Collapse button gets
+     * pressed - will only render a scroll bar if there are more
+     * than SCROLLBAR_MIN_LIMIT lines in file.
+     *
+     * (This method is bound to onClick for the Expand/Collapse button)
+     */
+    updateTextContent = () => {
+        let text = this.state.originalText;
+
+        if(this.state.expandButtonPushed){
+            let lines = text.split('\n');
+            let length = lines.length
+
+            if(lines.pop() === ""){
+                text = text + "<newline>"
+                lines[length-1] = "<newline>"
+             }
+
+            let display = this.getCollapsedText(lines);
+            this.setState({
```

Comment:
Strictly I don't think that you should rely on `this.state` when calling `this.setState` - from the React docs, setState updates may be asynchonous (i.e `this.state` may not be immediately updated after calling `this.setState`). I think the effect of this would be that if someone clicked the expand/collapse button _really_ quickly, not all clicks would register correctly (and possible lead to some weird inconsistent state).

https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

The docs suggest a form where you pass a callback that takes state and props as args - maybe we could use that form instead? (JS loves callbacks after all)